aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/RealAWSCognitoAuthPluginTest.kt

```diff
@@ -256,4 +283,266 @@ class RealAWSCognitoAuthPluginTest {
         // THEN
         coVerify { anyConstructed<ResetPasswordUseCase>().execute(username, options, onSuccess, onError) }
     }
+
+    @Test
+    fun `test signup API with given arguments`() {
+        // GIVEN
+        val username = "user"
+        val password = "password"
+        val email = "user@domain.com"
+        val options = AuthSignUpOptions.builder().userAttribute(AuthUserAttributeKey.email(), email).build()
+
+        val currentAuthState = mockk<AuthState> {
+            coEvery { authNState } returns AuthenticationState.Configured()
+        }
+        coEvery { authStateMachine.getCurrentState(captureLambda()) } coAnswers {
+            lambda<(AuthState) -> Unit>().invoke(currentAuthState)
+        }
+
+        val requestCaptor = slot<SignUpRequest.Builder.() -> Unit>()
+        coJustRun { authService.cognitoIdentityProviderClient?.signUp(capture(requestCaptor)) }
+
+        val expectedRequest: SignUpRequest.Builder.() -> Unit = {
+            clientId = "app Client Id"
+            this.username = username
+            this.password = password
+            userAttributes = listOf(
+                AttributeType {
+                    name = "email"
+                    value = email
+                }
+            )
+            secretHash = "dummy Hash"
+        }
+
+        // WHEN
+        plugin.signUp(username, password, options, {}, {})
+        Thread.sleep(1_000)
```

Comment:
We should try to avoid timed sleeps whenever possible. Look into runBlocking or latch here. Same for all other areas sleep is used in this test class.